### PR TITLE
fix: [CDS-75177]: Improve Execution logs for Jenkins to show url (and params)

### PR DESCRIPTION
**Pull request by ayushitiwari024**
## Describe your changes
feat: [CDS-75177](https://harness.atlassian.net/browse/CDS-75177?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ): Improve Execution logs for Jenkins to show url (and params)

Issue: Improve Execution logs for Jenkins to show url (and params)

Changes: Concatenated jenkins URL with the error message and saved it in the logger.

Testing: Tested this case locally: https://harness.atlassian.net/browse/CDS-75177

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

[CDS-79514]: https://harness.atlassian.net/browse/CDS-79514?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/53379)

[CDS-75177]: https://harness.atlassian.net/browse/CDS-75177?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ayushitiwari024** [comment]
trigger messagecheck

File: 960-api-services/src/main/java/io/harness/artifacts/jenkins/service/JenkinsRegistryUtils.java [hunk 1]
```diff
@@ -661,7 +662,9 @@ public QueueReference trigger(
       String jobName, JenkinsInternalConfig jenkinsInternalConfig, Map<String, String> parameters) throws IOException {
     Job job = getJob(jobName, jenkinsInternalConfig);
     if (job == null) {
-      throw new ArtifactServerException("No job [" + jobName + "] found", USER);
+      throw new ArtifactServerException("No job [" + jobName + "] found"
+              + " " + jenkinsInternalConfig.getJenkinsRegistryUrl(),
```
Comment (shivamnegi94) [inline 960-api-services/src/main/java/io/harness/artifacts/jenkins/service/JenkinsRegistryUtils.java:666]: We are suppose to show the Jenkins execution url in consoles log and jenkinsInternalConfig.getJenkinsRegistryUrl() will only return Jenkins Url , not the execution Job Url

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/artifacts/jenkins/JenkinsArtifactTaskHandler.java [hunk 2]
```diff
@@ -209,6 +209,7 @@ public ArtifactTaskExecutionResponse triggerBuild(
     try {
       JenkinsInternalConfig jenkinsInternalConfig =
           JenkinsRequestResponseMapper.toJenkinsInternalConfig(attributesRequest);
+      msg = msg + " " + jenkinsInternalConfig.getJenkinsUrl();
```
Comment (shivamnegi94) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/task/artifacts/jenkins/JenkinsArtifactTaskHandler.java:212]: Why we are not adding it to console logs ?

**Comment by shivamnegi94** [comment]
Please add Unit test  cases

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/artifacts/jenkins/JenkinsArtifactTaskHandler.java (hunk 2 above)
Comment (ayushitiwari024) [reply 930-delegate-tasks/src/main/java/io/harness/delegate/task/artifacts/jenkins/JenkinsArtifactTaskHandler.java:212]: We are doing it in line 244 
executionLogCallback.saveExecutionLog(msg + e, LogLevel.ERROR);

File: 960-api-services/src/main/java/io/harness/artifacts/jenkins/service/JenkinsRegistryUtils.java (hunk 1 above)
Comment (ayushitiwari024) [reply 960-api-services/src/main/java/io/harness/artifacts/jenkins/service/JenkinsRegistryUtils.java]: addressed

**Review by shivamnegi94 — APPROVED** [review]

**Review by prashantsharma04 — APPROVED** [review]

**Comment by prashantsharma04** [comment]
trigger uts

**Comment by ayushitiwari024** [comment]
trigger ut2

**Comment by ayushitiwari024** [comment]
trigger uts

**Comment by ayushitiwari024** [comment]
trigger review

**Pull request merged**